File: car-backend-ms/app/schemas/photo.py
from datetime import datetime

from pydantic import BaseModel

from app.db.models.photo import OperationType


class PhotoResponse(BaseModel):
    id: int
    user_id: int
    original_filename: str
    operation_type: OperationType
    operation_params: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class PhotoListResponse(BaseModel):
    photos: list[PhotoResponse]
    total: int

File: car-frontend/src/types/photo.ts
export type OperationType = 'car_segmentation' | 'car_part_segmentation' | 'edit_photo';
export type PhotoStatus = 'preview' | 'generating' | 'completed';

export interface PhotoResponse {
  id: number;
  user_id: number;
  original_filename: string;
  operation_type: OperationType;
  status: PhotoStatus;
  operation_params: Record<string, unknown> | null;
  created_at: string;
}

export interface PhotoListResponse {
  photos: PhotoResponse[];
  total: number;
}

